tests/typecheck/contrib/auth/test_decorators.yml

```
@@ -41,6 +41,28 @@
         from django.contrib.auth.decorators import user_passes_test
         @user_passes_test  # E: Argument 1 to "user_passes_test" has incompatible type "Callable[[HttpRequest], HttpResponse]"; expected "Callable[[Union[AbstractBaseUser, AnonymousUser]], bool]"
         def view_func(request: HttpRequest) -> HttpResponse: ...
+-   case: user_passes_test_has_user_of_type_auth_user_model
+    disable_cache: true
+    main: |
+        from typing import Union
+        from django.http import HttpRequest, HttpResponse
+        from django.contrib.auth.models import AnonymousUser
+        from django.contrib.auth.decorators import user_passes_test
+        from myapp.models import MyUser
+
+        def check_user(user: Union[MyUser, AnonymousUser]) -> bool: return True
+        @user_passes_test(check_user)
+        def view_func(request: HttpRequest) -> HttpResponse: ...
```

Can we check a couple of the often used and now plugin changed signatures from `django.contrib.auth`, e.g. `authenticate` and `login`

```suggestion
        from typing import Union
        from django.http import HttpRequest, HttpResponse
        from django.contrib.auth import authenticate, login, get_user
        from django.contrib.auth.models import AnonymousUser
        from django.contrib.auth.decorators import user_passes_test
        from myapp.models import MyUser
        
        reveal_type(authenticate)  # N: Revealed type is "def (...) -> myapp.models.MyUser | None"
        reveal_type(login)  # N: Revealed type is "..."
        reveal_type(get_user)  # N: Revealed type is "..."

        def check_user(user: Union[MyUser, AnonymousUser]) -> bool: return True
        @user_passes_test(check_user)
        def view_func(request: HttpRequest) -> HttpResponse: ...
```

We could perhaps change the name of the test to something like `test_auth_user_model` in that case? Also, its "sibling" test is in `models/test_contrib_models.yml` but I'm totally fine to keep this test under `contrib/auth/`